src/main/scala/com/fulcrumgenomics/rnaseq/CollectErccMetrics.scala

```diff
@@ -0,0 +1,294 @@
+    }
+
+    // require that the SAM/BAM header contains all the ERCC transcripts
+
+    val missingErccTranscripts = erccData.keySet -- in.header.getSequenceDictionary.getSequences.map(_.getSequenceName)
+    if (missingErccTranscripts.nonEmpty) {
+      throw new IllegalArgumentException(s"Missing ERCC transcripts in the SAM/BAM header: ${missingErccTranscripts.mkString(", ")}")
+    }
+
+    var totalReads: Long    = 0 // the total # of reads in the BAM file
+    var erccReads: Long     = 0 // the total # of reads that are counted as mapped to an ERCC transcript
+    var erccReadPairs: Long = 0 // the total # of read pairs that are counted as mapped to an ERCC transcript
+    val erccReadPairCounter = new SimpleCounter[String]() // the count of read pairs mapped to a given ERCC transcript
+    in.iterator
+      .filterNot { r => progress.record(r); r.secondary || r.supplementary } // no secondary or supplementary
+      .filter { r =>
+        totalReads += 1
+        // consider only records mapped to an ERCC transcript
+        r.mapped && r.mapq >= minimumMappingQuality && erccData.contains(r.refName)
+      }
+      .filter { r =>
+        erccReads += 1
+        // consider only paired records where both ends map to the sam ERCC transcript
+        r.paired && r.mapped && r.mateMapped && r.refIndex == r.mateRefIndex && r.firstOfPair
+      }
+      .foreach { r =>
+        erccReadPairs += 1
+        erccReadPairCounter.count(r.refName)
+      }
+    in.safelyClose()
+
+    // Transform to log2 counts to compute some simple correlation and linear regression metrics.
+    val counts            = ArrayBuffer[Double]()
+    val normalizedCounts  = ArrayBuffer[Double]()
+    val concentrations    = ArrayBuffer[Double]()
+    val maximumErccLength = erccData.keys.map { name => in.header.getSequence(name).getSequenceLength }.max
+    erccData.iterator.foreach { case (name, concentration) =>
+      val count = erccReadPairCounter.countOf(name)
+      if (minTranscriptCount <= count) {
+        val erccLength = in.header.getSequence(name).getSequenceLength
+        counts += log2(count)
+        normalizedCounts += log2(count * maximumErccLength / erccLength.toDouble) // normalize by the ERCC transcript length
+        concentrations += log2(concentration)
+      }
+    }
+
+    // Rocket science
+    val simpleRegression = new SimpleRegression()
+    normalizedCounts.zip(concentrations).foreach { case (cnt, conc) => simpleRegression.addData(cnt, conc) }
+
+    val erccMetrics = ErccSummaryMetrics(
+      total_reads                = totalReads,
+      ercc_reads                 = erccReads,
+      fraction_ercc_reads        = if (totalReads == 0) 0 else 100.0 * erccReads / totalReads.toDouble,
+      ercc_templates             = erccReadPairs,
+      total_transcripts          = erccReadPairCounter.iterator.length,
+      total_filtered_transcripts = erccReadPairCounter.iterator.count(_._2 >= minTranscriptCount),
```

Comment:
`passing_filter_transcripts`?  I struggle with `filtered` because it's never clear to me whether it means things that were filtered _out_, or things kept after filtering.